src/nvim/option.c

```diff
@@ -4095,9 +4095,21 @@ static char *set_bool_option(const int opt_idx, char_u *const varp,
       curbuf->b_p_iminsert = B_IMODE_NONE;
       curbuf->b_p_imsearch = B_IMODE_USE_INSERT;
     }
+  } else if (((int *)varp == &curwin->w_p_nu
+              || (int *)varp == &curwin->w_p_rnu)
+             && (*curwin->w_p_scl == 'n' && *(curwin->w_p_scl + 1) == 'u')
+             && curbuf->b_signlist != NULL) {
+    // If the 'number' or 'relativenumber' options are modified and
+    // 'signcolumn' is set to 'number', then clear the screen for a full
+    // refresh. Otherwise the sign icons are not displayed properly in the
+    // number column.  If the 'number' option is set and only the
+    // 'relativenumber' option is toggled, then don't refresh the screen
+    // (optimization).
+    if (!(curwin->w_p_nu && ((int *)varp == &curwin->w_p_rnu))) {
+      redraw_all_later(CLEAR);
```

Comment:
NOT_VALID

Reply:
I should have noticed that. Perhaps we should add a lint rule.

Reply:
Though perhaps this is at all NA, because we do not have sign icons (i e pictures drawn by gnvim) ? Has any problem actually been noticed in nvim? If we ever add icons, we should make sure they can be erased just by drawing a space char over them.

Reply:
> Has any problem actually been noticed in nvim?

No.

Sounds like N/A then indeed.
I've thought it would apply to the signcolumn in general - but the initial/first patch (https://github.com/vim/vim/commit/2b044ffb5ada77e6fa89779d6532ea9fae3fe029) is for gvim already only, right?